File: sdk/dart/lib/src/admin_service.dart
import 'client.dart';
import 'types.dart';

/// Superuser (admin panel) session management.
class AdminService {
  final CratebaseClient client;

  AdminService(this.client);

  Future<AdminAuthResponse> authWithPassword(String email, String password) async {
    final json = await client.send(
      '/api/admins/auth-with-password',
      method: 'POST',
      body: {'email': email, 'password': password},
    );
    final result = AdminAuthResponse.fromJson(json as Map<String, dynamic>);
    client.authStore.save(result.token, result.admin.toJson());
    return result;
  }

  Future<AdminAuthResponse> authRefresh() async {
    final json = await client.send('/api/admins/auth-refresh', method: 'POST');
    final result = AdminAuthResponse.fromJson(json as Map<String, dynamic>);
    client.authStore.save(result.token, result.admin.toJson());
    return result;
  }
}

File: sdk/js/src/admin-service.ts
import type { Cratebase } from "./client.js";
import type { AdminAuthResponse } from "./types.js";

/** Superuser (admin panel) session management. */
export class AdminService {
  constructor(private readonly client: Cratebase) {}

  async authWithPassword(email: string, password: string): Promise<AdminAuthResponse> {
    const result = await this.client.send<AdminAuthResponse>("/api/admins/auth-with-password", {
      method: "POST",
      body: { email, password },
    });
    this.client.authStore.save(result.token, result.admin);
    return result;
  }

  async authRefresh(): Promise<AdminAuthResponse> {
    const result = await this.client.send<AdminAuthResponse>("/api/admins/auth-refresh", { method: "POST" });
    this.client.authStore.save(result.token, result.admin);
    return result;
  }
}
